### DISCUSSION TITLE
October CMS upgrade: composer update command failed. Can't update the project

### DISCUSSION THREAD

**Discussion by 01Kuzma** [wintercms/winter#832] [Q&A] [answered]
Hello!
I have October CMS build # 476. The first build was installed locally using the wizard and then copied to shared hosting. Then the update process was running on hosting.
Now, I'd like to upgrade it to Winter CMS.
I've copied it to Windows localhost (PHP 8.1.10 + latest composer).
I'm a total newbie regarding composer, so I've copied this [composer.json](https://github.com/wintercms/winter/blob/develop/composer.json) file to root of the project and added `platform`, because running `composer update` command caused many `Root composer.json requires php ^8.0.2 but your php version (7.4.33) does not satisfy that requirement.` errors. So,  "platform looks like":

```
    "platform": {
        "php": "8.0.2"
    }		
    }
```

After I've run the command I get these errors:

```
Generating autoload files
85 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
> @php artisan winter:version
Missing vendor files, try running "composer install" or use the Wizard installer.
Script @php artisan winter:version handling the post-update-cmd event returned with error code 1
```

Project is not starting and I'm getting `Missing vendor files, try running "composer install" or use the Wizard installer. ` error.

What should be done further co complete the upgrade process or it should be start over again?
Need a step by step advice.

Thank you!

**Comment by LukeTowers** [comment, marked as answer]
@01Kuzma See https://wintercms.com/blog/post/how-start-using-winter-cms, your specific issues are twofold:

1. You need to switch the command line version of PHP you're using to at least PHP 8, using the platform section in composer.json is a temporary hack that will cause other issues
2. You need to update the `bootstrap/autoload.php` file.

I would highly recommend following that guide as well as the upgrade guide to Laravel 6 / Winter v1.1 & the upgrade guide to Laravel 9 / Winter v1.2 if your application is complex and or has significant business value to you. If it's just a basic brochure style site then you might even find it easiest to just install a new copy of the latest Winter CMS codebase and copy over your plugins & themes folders. 

I would recommend replacing any RainLab plugins with their Winter equivalents though.

**Comment by 01Kuzma** [comment]
Thank you!
The first part with the command line has been fixed, I had `PHP 8` by default installed, but something caused `composer` to "think", that it was a lower version.
I'll try to finish the upgrade process. Earlier I tried to rebuild the app from the scratch, I succeeded, but my old app has many hundred posts with photos and I was unable to migrate them to a new version.

**Comment by 01Kuzma** [comment]
Two more questions :)
1. It seems that the composer upgrade process has been finished. 
Laravel version is 9.50.2 - Winter CMS, 
Following these [notes](https://github.com/wintercms/meta/blob/master/l6-upgrade-notes.md#upgrade-laravel-packages)
files has been added:
   [config/auth.php](https://github.com/wintercms/winter/blob/develop/config/auth.php)
    [config/develop.php](https://github.com/wintercms/winter/blob/develop/config/develop.php)
    [config/hashing.php](https://github.com/wintercms/winter/blob/develop/config/hashing.php)
    [config/logging.php](https://github.com/wintercms/winter/blob/develop/config/logging.php)

`.htaccess` edited also optional changes have been made.

The problem is that in the backend it's shown that the app version is still # 476, but the changelog shows 1.2.1
![image](https://user-images.githubusercontent.com/569504/216770011-0e3bf24b-f4bb-4b7d-9615-a9f88ce08f2b.png)
But on the main page the Winter Logo is shown:
![image](https://user-images.githubusercontent.com/569504/216770031-a656a45c-be0f-40cc-aa52-cbb122f2b2ef.png)


2. And what's the best way to upgrade the existing RainLab plugins to Winter? 


Thank you!

**Reply by bennothommo** [reply]
@01Kuzma don't worry too much about the build version in the Updates & Plugins screen. That's due to the marketplace not being available. You can run `php artisan winter:version` to see your true Winter CMS version (this will also update that build version in the backend if the database is available).

As for your second point, if you are using Composer, you can delete the `rainlab` directory from the `plugins` folder and add the Winter plugins as Composer dependencies, ie. `"winter/wn-blog-plugin": "^2.0.2"` in your `require` section, and so on.

**Comment by 01Kuzma** [comment]
@bennothommo , thank you!
The command gives this info: "We could not accurately determine your Winter CMS build due to the number of modifications. The closest detected build is Winter CMS build 1.2.1."
So, I assume that the upgrade process is successful.

But if I will remove the `rainlab` directory and add winter plugins, for example `blog`, the data of blog posts and etc. will be lost?

**Reply by LukeTowers** [reply]
The command says that because you're running on the develop branch and there hasn't been a new release in a while.

As long as you just replace the plugin files and don't run any commands like plugin:remove the data should be picked up and migrated by the winter plugins. Take a backup though, it's possible that your rainlab plugin versions could be newer than the max supported by winter replacements, if that's the case they won't be auto migrated and you can submit issues for each plugin to bump their version

**Reply by 01Kuzma** [reply]
Unfortunately, the first attempt to upgrade Rainlab's Blog plugin to Winter [failed](https://github.com/wintercms/wn-blog-plugin/issues/33) :(
